### ISSUE TITLE
bug: incorrect detection of an empty line in "rule-properties-order"

### ISSUE THREAD

**Issue by volochiy-s** [stylelint/stylelint#682]
I faced the problem where rule-properties-order reports "Expected an empty line..." when empty line is in place

Let's say, my CSS looks like this
```css
.test {
    position: relative;
    display: block;
}
```

and my stylelint.config.js is like

```javascript
module.exports  = {
    rules: {
        "rule-properties-order": [
            "position",

            {
                emptyLineBefore: true,
                properties: [
                    "display"
                ]
            }
        ]
    }
}
```

This code reports an issue - as expected, there are no empty line between 'position' and 'display'. I want to fix it. 

If I make a newline by hitting Enter with cursor positioned like on this picture - I will get an error notification.
![image](https://cloud.githubusercontent.com/assets/6917987/12777754/8ae38b6a-ca67-11e5-84df-312f6bf112e3.png)

But if I make a newline by hitting Enter in the two following cases - everything works fine
![image](https://cloud.githubusercontent.com/assets/6917987/12777737/7a351d38-ca67-11e5-9e6e-52c518a0821d.png)
![image](https://cloud.githubusercontent.com/assets/6917987/12777670/16549154-ca67-11e5-8377-701cd19852d0.png)

**Comment by davidtheclark** [comment]
@volochiy-s: Does this issue only occur when using your text editor? If you can reproduce it outside of your editor, please re-open this issue. But if it's a problem with the editor plugin, please file an issue with that plugin instead. Make sense?

**Comment by volochiy-s** [comment]
@davidtheclark: reproducible in Atom, Notepad++, Visual Studio Code.

Most likely it's because autoindent in editors: I have 4 spaces there, if I remove them, the error notification disappears.

I'd like to reopen this issue, but I have no rights.
thanks.

**Comment by davidtheclark** [comment]
@volochiy-s: Can you please post stringified versions of the CSS that is and is not causing the issue --- by "stringified" I mean what `JSON.stringify()` would produce, with `\n` and `\r\n` characters. Otherwise I'm not sure how I can write a test to reproduce this.

**Comment by volochiy-s** [comment]
@davidtheclark 
causing the issue: 
.test {\r\n    position: relative;\r\n    \r\n    display: block;\r\n}

not causing the issue:
.test {\r\n    position: relative;\r\n\r\n    display: block;\r\n}

**Comment by volochiy-s** [comment]
@davidtheclark 
causing the issue: 
`.test {\r\n    position: relative;\r\n    \r\n    display: block;\r\n}`

not causing the issue:
`.test {\r\n    position: relative;\r\n\r\n    display: block;\r\n}`

**Comment by volochiy-s** [comment]
Four spaces should be here. But it is still reproducible even with one
![image](https://cloud.githubusercontent.com/assets/6917987/12787521/6d1916cc-ca9e-11e5-834e-a7beff200927.png)

**Comment by davidtheclark** [comment]
Thanks! I think the bug will be easy to fix with that info.
